File: solutions/c/hamming/1/hamming.c
#include "hamming.h"

#include <string.h>

int compute(const char *lhs, const char *rhs) {
    int lhs_length = strlen(lhs);
    int rhs_length = strlen(rhs);

    if (lhs_length != rhs_length) return -1;
    
    int dist = 0;
    for (int i = 0; i < lhs_length; i++) {
        if (lhs[i] != rhs[i]) dist++;
    }

    return dist;
}

File: solutions/lua/hamming/1/hamming.lua
local Hamming = {}

function Hamming.compute(a, b)
    if #a ~= #b then
        return -1
    end

    local dist = 0

    for i = 1, #a do
        if string.sub(a, i, i) ~= string.sub(b, i, i) then
            dist = dist + 1
        end
    end

    return dist
end

return Hamming
